spacebarchat/server
# Add Initial Support for working webrtc (FC-Server)

Add Initial Support for working webrtc (FC-Server)

- Update packages to latest versions (including stresstest)
- Update clients to latest (both regular, and dev)
- Add future support for dev portal theming with CSS (and plugins)
- Fix broken GLOBAL_VARS in html's to match better with discord's version
- Fix broken WebRTC in latest fosscord refactor.
- Add HOSTNAME (and other) env vars, to work with WebRTC better with the browser (still would like it to get the IP from browser and work from any endpoint rather than just one)
- Change 127.0.0.1 & Localhost's to 0.0.0.0 to work better as a "server" NOTE: you need to enter http(s)://localhost:3001/ to your browser, as 0.0.0.0 is just for hosting, not viewing
- Based off of https://github.com/fosscord/fosscord-server/pull/472
- Remove the need for the webrtc.js plugin (whatever this even does anyways)
- Make Voice endpoint "/voice" instead of having to port forward more (by default)

THINGS THAT STILL NEED TO GET FIXED!:

- Video (and screen-share) doesn't work at the moment (client issue?)
- Defending your sound doesn't work, webrtc reconnects after muting, unmuting, defending, undefending for unknown reason.
- May stop chat from loading (slowdown possible, browser issue?)
- chrome://flags/#unsafely-treat-insecure-origin-as-secure doesn't exist on Firefox (don't require this for fc, might not be possible, only needed for http NOT https)
- You may hear yourself talk even if no one is there
- You tell me.

Admins, Please Note: I changed the 300x all back to 3001, but some were complaining they want it back to 300x, should I change its default (fallback) back?
